src/cache.c

```diff
@@ -9,19 +9,22 @@
  */
 struct cache_entry *alloc_entry(char *path, char *content_type, void *content, int content_length)
 {
-    ///////////////////
-    // IMPLEMENT ME! //
-    ///////////////////
+    struct cache_entry *e = malloc(sizeof(struct cache_entry));
```

Comment:
Do you have to malloc size of struct cache_entry here? Since cache_entry is already a struct, e is a struct, can you simplify your right side?